### PR TITLE
Mod option for shield belts

### PR DESCRIPTION
**Pull request by perkinslr**
## Additions

Add mod option to let shield belts keep working when using a manned turret.

## Changes

Let binoculars work while wearing shield belt

## Reasoning

CE needs to be balanced to be fun when used as the only mod, but needs to be easy enough to balance as part of a set of mods.  The default behavior works reasonably well when CE is used alone, but giving the player more options is rarely a bad idea.  (Given that it's a toggle switch, it's unlikely to cause analysis paralysis).

## Alternatives

Pick to either enable or disable the new option.
Use a sub mod to achieve this behavior.

## Testing

Check tests you have performed:
- [X] Compiles without warnings
- [X] Game runs without errors
- [X] Playtested a colony (minutes)

### CODE REVIEW

File: Languages/English/Keyed/ModMenu.xml [hunk 1]
```diff
@@ -23,6 +23,9 @@
   <CE_Settings_MergeExplosions_Title>Merge explosions</CE_Settings_MergeExplosions_Title>
   <CE_Settings_MergeExplosions_Desc>Nearby explosions are merged into eachother, significantly improving performance during a massive explosives cook-off. Strongly recommended to leave on.</CE_Settings_MergeExplosions_Desc>
 
+  <CE_Settings_TurretsBreakShields_Title>Turrets Break Shield Belts</CE_Settings_TurretsBreakShields_Title>
+  <CE_Settings_TurretsBreakShields_Desc>When pawns wearing shield belts use a turret, disable the shield belt until they stop.</CE_Settings_TurretsBreakShields_Desc>
```
Comment (N7Huntsman) [inline Languages/English/Keyed/ModMenu.xml:27]: Suggested replacement:
`When enabled, shield belts will deactivate while colonists are manning turrets. If disabled, shield belts will instead remain active, offering additional protection.`

File: Languages/English/Keyed/ModMenu.xml [hunk 2]
```diff
@@ -23,6 +23,9 @@
   <CE_Settings_MergeExplosions_Title>Merge explosions</CE_Settings_MergeExplosions_Title>
   <CE_Settings_MergeExplosions_Desc>Nearby explosions are merged into eachother, significantly improving performance during a massive explosives cook-off. Strongly recommended to leave on.</CE_Settings_MergeExplosions_Desc>
 
+  <CE_Settings_TurretsBreakShields_Title>Turrets Break Shield Belts</CE_Settings_TurretsBreakShields_Title>
```
Comment (N7Huntsman) [inline Languages/English/Keyed/ModMenu.xml:26]: Suggested replacement: `Disable shield belts on turrets`

**Review by N7Huntsman — CHANGES_REQUESTED** [review]

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Mod seems to function correctly, but I'd like to revise the text a bit to make it a little clearer.

**Review by N7Huntsman — APPROVED** [review]

File: Source/CombatExtended/CombatExtended/Settings.cs [hunk 3]
```diff
@@ -18,12 +18,14 @@ public class Settings : ModSettings
         private bool allowMeleeHunting = false;
         private bool smokeEffects = true;
         private bool mergeExplosions = true;
+	private bool turretsBreakShields = true;
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Settings.cs:21]: Please keep tabs/spaces consistent, at least within the same file (change tabs->spaces please)

File: Source/CombatExtended/CombatExtended/Settings.cs [hunk 4]
```diff
@@ -18,12 +18,14 @@ public class Settings : ModSettings
...
 
         public bool ShowCasings => showCasings;
         public bool ShowTaunts => showTaunts;
         public bool AllowMeleeHunting => allowMeleeHunting;
         public bool SmokeEffects => smokeEffects;
         public bool MergeExplosions => mergeExplosions;
+	public bool TurretsBreakShields => turretsBreakShields;
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Settings.cs:28]: Same here

File: Source/CombatExtended/CombatExtended/Settings.cs [hunk 5]
```diff
@@ -75,6 +77,7 @@ public override void ExposeData()
             Scribe_Values.Look(ref allowMeleeHunting, "allowMeleeHunting", false);
             Scribe_Values.Look(ref smokeEffects, "smokeEffects", true);
             Scribe_Values.Look(ref mergeExplosions, "mergeExplosions", true);
+	    Scribe_Values.Look(ref turretsBreakShields, "turretsBreakShields", true);
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/Settings.cs:80]: and here

**Review by MadaraUchiha — APPROVED** [review]
LGTM, though please maintain code formatting :)

File: Source/CombatExtended/CombatExtended/Settings.cs (hunk 5 above)
Comment (perkinslr) [reply Source/CombatExtended/CombatExtended/Settings.cs]: Not sure how that happened, my editor was set to try to match the whitespace rules of existing files but got itself confused.  I've got it set to force spaces now.  

At some point we should write a `clangformat` spec for the C# and run it on the whole repo, as the style varies a bit from file to file.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**